test/prerequisite-tasks.js

```diff
@@ -0,0 +1,247 @@
+	execaStub.createStub([
+		{
+			command: 'git ls-remote origin HEAD',
+			exitCode: 1,
+			exitCodeName: 'EPERM',
+			stderr: 'not found'
+		}
+	]);
+	await t.throwsAsync(run(testedModule('2.0.0', {name: 'test', version: '1.0.0'}, {yarn: false})),
+		{message: 'not found'});
+	t.true(tasks.some(task => task.title === 'Check git remote' && task.hasFailed()));
+});
+
+test.serial('should fail when version is invalid', async t => {
+	await t.throwsAsync(run(testedModule('DDD', {name: 'test', version: '1.0.0'}, {yarn: false})),
+		{message: `Version should be either ${version.SEMVER_INCREMENTS.join(', ')}, or a valid semver version.`});
+	t.true(tasks.some(task => task.title === 'Validate version' && task.hasFailed()));
+});
+
+test.serial('should fail when version is lower as latest version', async t => {
+	await t.throwsAsync(run(testedModule('0.1.0', {name: 'test', version: '1.0.0'}, {yarn: false})),
+		{message: 'New version `0.1.0` should be higher than current version `1.0.0`'});
+	t.true(tasks.some(task => task.title === 'Validate version' && task.hasFailed()));
+});
+
+test.serial('should fail when prerelease version of public package without dist tag given', async t => {
+	await t.throwsAsync(run(testedModule('2.0.0-1', {name: 'test', version: '1.0.0'}, {yarn: false})),
+		{message: 'You must specify a dist-tag using --tag when publishing a pre-release version. This prevents accidentally tagging unstable versions as "latest". https://docs.npmjs.com/cli/dist-tag'});
+	t.true(tasks.some(task => task.title === 'Check for pre-release version' && task.hasFailed()));
+});
+
+test.serial('should not fail when prerelease version of public package with dist tag given', async t => {
+	execaStub.createStub([
+		{
+			command: 'git rev-parse --quiet --verify refs/tags/v2.0.0',
+			stdout: ''
+		}
+	]);
+	await run(testedModule('2.0.0-1', {name: 'test', version: '1.0.0'}, {yarn: false, tag: 'pre'}));
+	t.pass();
+});
+
+test.serial('should not fail when prerelease version of private package without dist tag given', async t => {
+	execaStub.createStub([
+		{
+			command: 'git rev-parse --quiet --verify refs/tags/v2.0.0',
+			stdout: ''
+		}
+	]);
+	await run(testedModule('2.0.0-1', {name: 'test', version: '1.0.0', private: true}, {yarn: false}));
+	t.pass();
+});
+
+test.serial('should fail when git tag already exists', async t => {
+	execaStub.createStub([
+		{
+			command: 'git rev-parse --quiet --verify refs/tags/v2.0.0',
+			stdout: 'vvb'
+		}
+	]);
+	await t.throwsAsync(run(testedModule('2.0.0', {name: 'test', version: '1.0.0'}, {yarn: false})),
+		{message: 'Git tag `v2.0.0` already exists.'});
+	t.true(tasks.some(task => task.title === 'Check git tag existence' && task.hasFailed()));
+});
+
+test.serial('checks should pass', async t => {
+	execaStub.createStub([
+		{
+			command: 'git rev-parse --quiet --verify refs/tags/v2.0.0',
+			stdout: ''
+		}
+	]);
+	await run(testedModule('2.0.0', {name: 'test', version: '1.0.0'}, {yarn: false}));
```

Comment:
This should use `notThrowsAsync` instead of `t.pass()`.

Reply:
Done